samples/dynamodb/lambda/DynamoDB_1/Sales/OrderSaga.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Threading.Tasks;
+
+using NServiceBus;
+using NServiceBus.Logging;
+
+#region OrderSaga
+
+public class OrderSaga : Saga<OrderSagaData>,
+  IAmStartedByMessages<PlaceOrder>,
+  IHandleMessages<CustomerBilled>,
+  IHandleMessages<InventoryStaged>,
+  IHandleTimeouts<CancelOrder>
+{
+  static ILog log = LogManager.GetLogger(typeof(OrderSagaData));
+
+  protected override void ConfigureHowToFindSaga(SagaPropertyMapper<OrderSagaData> mapper)
+  {
+    mapper.MapSaga(sagaData => sagaData.OrderId)
+     .ToMessage<PlaceOrder>(s => s.OrderId)
+     .ToMessage<CustomerBilled>(s => s.OrderId)
+     .ToMessage<InventoryStaged>(s => s.OrderId)
+     .ToMessage<CancelOrder>(s => s.OrderId);
+  }
+
+  public async Task Handle(PlaceOrder message, IMessageHandlerContext context)
+  {
+    log.Info($"Placing order: {Data.OrderId}");
+
+    await context.Publish(new OrderReceived()
+    {
+      OrderId = message.OrderId
+    });
+
+    // Delay 5 seconds to allow time for order cancellation
+    await RequestTimeout<CancelOrder>(context, DateTimeOffset.UtcNow.AddSeconds(5));
```

Comment:
I think we should remove the whole cancellation path completely to simplify the sample. Currently it's quite tricky to get a good timings to ensure that there are both successful orders and cancellations happening at a reasonable distribution. But in the end, it doesn't really end up providing much value in the context of the sample so why not just remove it?

Reply:
I'm good with that. Simpler is better.